### ISSUE TITLE
Integer values are not parsed correctly

### ISSUE THREAD

**Issue by keul** [gorilla-co/odata-query#9]
I'm using the raw lexer (no provided mongodb or other integrations).

Let's look at this example:

```python
from odata_query.grammar import ODataLexer, ODataParser
lexer = ODataLexer()
parser = ODataParser()
odata_filter = parser.parse(lexer.tokenize("foo eq 15"))
```

I would expect that `odata_filter` contains a numerical value for `foo`, instead I get:

```
Compare(comparator=Eq(), left=Identifier(name='foo'), right=Integer(val='15'))
```

**Comment by OliverHofkens** [comment]
Hey @keul ,
I think your example might be missing something as I don't see what you mean with `SomethingElse`.
The parser will always output an `AST`, and it's up to the ORM integration or a custom transormer to turn that into a usable query. So the `Compare` node you see is exactly what I'd expect for that query.

**Comment by keul** [comment]
Really sorry @OliverHofkens, opened the issue while rushing (`SomethingElse` was `foo`).

My question was about doing this:

```
>>> odata_filter.right
Integer(val='15')
>>> odata_filter.right.val
'15'
```

There's not an (easy) way to obtain the Python value expected from this token (so the `15` integer value and not the `'15'` str value).

I can obviously cast it myself by looking at the token type (`Integer` in this case) but I'm wondering if there's a more straightforward way.

**Comment by keul** [comment]
Sure, I see the point!

Adding a new property getter with converted values would be great for a end-user point of view IMHO.

**Comment by OliverHofkens** [comment]
I just released `v0.5` which includes this change!
`Literal` nodes now have a `py_val` property, which is the closest Python equivalent of the OData type.
I refactored the existing `Transformer`s to use this new property as well.

Note that release 0.5 also includes full typing support, so keep that in mind if you use Mypy and upgrade.

Thanks for suggesting this!
